src/NodeAnalyzer/PropertyFetchAnalyzer.php

```diff
@@ -104,6 +104,24 @@ public function containsLocalPropertyFetchName(Trait_ $trait, string $propertyNa
         );
     }
 
+    public function containsWrittenPropertyFetchName(Trait_ $trait, string $propertyName): bool
+    {
+        if ($trait->getProperty($propertyName) instanceof Property) {
+            return true;
+        }
+
+        return (bool) $this->betterNodeFinder->findFirst(
+            $trait,
+            function (Node $node) use ($propertyName): bool {
+                if (! $node instanceof Assign) {
```

Comment:
This should cover assign op, eg `$this->props .= "append";` as well, using attribute key is_being_assigned should be more reliable.

https://github.com/rectorphp/rector-src/blob/c59c83a99fdb883dbcf6fead62b7cd9961fae034/src/NodeManipulator/AssignManipulator.php#L64

Also needs to verify if property is being `unset()` via trait.

, i will check it.

Reply:
https://github.com/rectorphp/rector-src/pull/6664